packages/components/popover/src/use-popover.ts

```diff
@@ -174,6 +179,16 @@ export function usePopover(originalProps: UsePopoverProps) {
     state,
   );
 
+  const placement = useMemo(() => {
+    // If ariaPlacement is null, popoverProps.style isn't set,
+    // so we return null to avoid an incorrect animation value.
+    if (!ariaPlacement) {
+      return null;
+    }
+
+    return getShouldUseAxisPlacement(ariaPlacement, placementProp) ? ariaPlacement : placementProp;
+  }, [ariaPlacement, placementProp]);
```

Comment:
Incorrect initial placement caused animation to be applied in the wrong direction.